In [1]:
import pandas as pd
import plotly.express as px

# Load the dataset
data = pd.read_csv('/Users/manishghoshal/Downloads/Futurense_Internship-main/Final Internship Capstone Project/Financial Analytics/financialanalytics.csv')

In [2]:
# Calculate the total profit for each company
data['Total_Profit'] = data.groupby('Company')['Profit'].transform('sum')

In [3]:
# Find the company with the highest total profit
highest_profit_company = data.loc[data['Total_Profit'] == data['Total_Profit'].max(), 'Company'].values[0]
highest_profit_value = data.loc[data['Company'] == highest_profit_company, 'Total_Profit'].values[0]
print(f"The company with the highest total profit is {highest_profit_company} with a total profit of {highest_profit_value}")

The company with the highest total profit is C with a total profit of 12271563


In [4]:
# Calculate the average profit for each industry
avg_profit_by_industry = data.groupby('Industry')['Profit'].mean().reset_index()
print("Average profit by industry:")
print(avg_profit_by_industry)

Average profit by industry:
     Industry        Profit
0     Finance  46035.342105
1  Healthcare  43040.506098
2  Technology  44124.124242


In [5]:
# Visualize the total profit for each company
fig = px.bar(data, x='Company', y='Total_Profit', color='Industry',
             title='Total Profit by Company')
fig.update_layout(xaxis_title='Company', yaxis_title='Total Profit')
fig.show()


In [6]:
# Visualize the average profit by industry
fig = px.bar(avg_profit_by_industry, x='Industry', y='Profit',
             title='Average Profit by Industry')
fig.update_layout(xaxis_title='Industry', yaxis_title='Average Profit')
fig.show()

**Infer findings:**

1. **Company with Highest Total Profit:**
   - The company 'C' has the highest total profit of 12271563.

2. **Average Profit by Industry:**
   - Industry: Finance, Average Profit: 46035.34
   - Industry: Healthcare, Average Profit: 43040.51
   - Industry: Technology, Average Profit: 44124.12
